src/Core/Types.Sorting/SortInputTypeDescriptor.cs

```diff
@@ -0,0 +1,145 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Reflection;
+using HotChocolate.Types.Descriptors;
+using HotChocolate.Types.Descriptors.Definitions;
+using HotChocolate.Utilities;
+
+namespace HotChocolate.Types.Sorting
+{
+    public class SortInputTypeDescriptor<T>
+        : DescriptorBase<SortInputTypeDefinition>
+        , ISortInputTypeDescriptor<T>
+    {
+        protected SortInputTypeDescriptor(
+            IDescriptorContext context,
+            Type entityType)
+            : base(context)
+        {
+            if (entityType is null)
+            {
+                throw new ArgumentNullException(nameof(entityType));
+            }
+
+            Definition.EntityType = entityType;
+            Definition.ClrType = typeof(object);
+            Definition.Name = context.Naming.GetTypeName(
+                entityType, TypeKind.Object) + "Sort";
```

Comment:
@PascalSenn what do you think about the post fix here... with the filter inputs we have something like BarFilter.... BarSort kinda sound strange... BarSortClause .... BarSortInput.... ?? But this would also be strange with BarFilter being different to Sort.... any ideas?